_search-plugins/sql/ppl/subsearch.md

```diff
@@ -0,0 +1,162 @@
+---
+layout: default
+title: Subsearch
+parent: PPL
+grand_parent: SQL and PPL
+nav_order: 3
+---
+
+# PPL subsearch (experimental)
+
+The `subquery` (aka subquery) contain 4 types: `in`, `exists`, `scalar` and `relation`. The first three are expressions, they are used in `where` command (`where <boolean expression>`) and search filter (`search source=* <boolean expression>`). The last one is a statement.
+
+## In subsearch
+
+### Syntax
+```sql
+where <field> [not] in [ search source=... | ... | ... ]
+```
+
+### Usage
+```sql
+source = outer | where a in [ source = inner | fields b ]
+source = outer | where (a) in [ source = inner | fields b ]
+source = outer | where (a,b,c) in [ source = inner | fields d,e,f ]
+source = outer | where a not in [ source = inner | fields b ]
+source = outer | where (a) not in [ source = inner | fields b ]
+source = outer | where (a,b,c) not in [ source = inner | fields d,e,f ]
+source = outer a in [ source = inner | fields b ] // search filtering with subquery
+source = outer a not in [ source = inner | fields b ] // search filtering with subquery)
+source = outer | where a in [ source = inner1 | where b not in [ source = inner2 | fields c ] | fields b ] // nested
+source = table1 | inner join left = l right = r on l.a = r.a AND r.a in [ source = inner | fields d ] | fields l.a, r.a, b, c //as join filter
+```
+
+## Exists subsearch
+
+### Syntax
+```sql
+where [not] exists [ search source=... | ... | ... ]
+```
+
+### Usage
+```sql
+// Assumptions: `a`, `b` are fields of table outer, `c`, `d` are fields of table inner,  `e`, `f` are fields of table nested
+source = outer | where exists [ source = inner | where a = c ]
+source = outer | where not exists [ source = inner | where a = c ]
+source = outer | where exists [ source = inner | where a = c and b = d ]
+source = outer | where not exists [ source = inner | where a = c and b = d ]
+source = outer exists [ source = inner | where a = c ] // search filtering with subquery
+source = outer not exists [ source = inner | where a = c ] //search filtering with subquery
+source = table as t1 exists [ source = table as t2 | where t1.a = t2.a ] //table alias is useful in exists subquery
+source = outer | where exists [ source = inner1 | where a = c and exists [ source = nested | where c = e ] ] //nested
+source = outer | where exists [ source = inner1 | where a = c | where exists [ source = nested | where c = e ] ] //nested
+source = outer | where exists [ source = inner | where c > 10 ] //uncorrelated exists
+source = outer | where not exists [ source = inner | where c > 10 ] //uncorrelated exists
+source = outer | where exists [ source = inner ] | eval l = "nonEmpty" | fields l //special uncorrelated exists
```

Comment:
what is use case? check if the inner table has any rows (without correlation)?
why include `eval l = "nonEmpty"`?